adapters/element/index.js

```diff
@@ -0,0 +1,129 @@
+            {
+                headers: {
+                    "User-Agent":
+                        "Mozilla/5.0 (Macintosh; Intel Mac OS X 11_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4619.141 Safari/537.36",
+                },
+            },
+        );
+        const symbol = resp.data;
+        return symbol;
+    };
+    getPrice = async timestamp => {
+        const resp = await axios.get(
+            `${URL}/token-price?key=${KEY}&token_address=${this.token}&protocol=${this.protocol}&timestamp=${timestamp}`,
+            {
+                headers: {
+                    "User-Agent":
+                        "Mozilla/5.0 (Macintosh; Intel Mac OS X 11_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4619.141 Safari/537.36",
+                },
+            },
+        );
+        return resp.data;
+    };
+    stop = async () => {
+        this.sdk.stop();
+    };
+
+    // getBuyer = async event => {
+    //     const buyer = event.returnValues.owner;
+    //     if (event.event === "NameRenewed") {
+    //         const txReceipt = await this.sdk.getTransactionReceipt(event.transactionHash);
+    //         if (txReceipt === null) {
+    //             return null;
+    //         }
+    //         return txReceipt.from;
+    //     }
+    //     return buyer;
+    // };
+
+    process = async event => {
+        const isER721 = event.event === "ERC721SellOrderFilled" || event.event === "ERC721BuyOrderFilled";
+        const isSellOrder = ["ERC721SellOrderFilled", "ERC1155SellOrderFilled"].includes(event.event);
+        const block = await this.sdk.getBlock(event.blockNumber);
+        const timestamp = moment.unix(block.timestamp).utc();
+        const po = await this.getPrice(block.timestamp);
+        const amount = isER721 ? 1 : event.returnValues["erc1155FillAmount"];
+        const price = isER721 ? event.returnValues["erc20TokenAmount"] : event.returnValues["erc20FillAmount"];
+        const nativePrice = new BigNumber(price).dividedBy(10 ** this.symbol.decimals);
+        const maker = event.returnValues["maker"];
+        const taker = event.returnValues["taker"];
+        const buyer = isSellOrder ? taker : maker;
+        const seller = isSellOrder ? maker : taker;
+        const nft_contract = event.returnValues["erc721Token"] || event.returnValues["erc1155Token"];
+        const tokenId = event.returnValues["erc721TokenId"] || event.returnValues["erc1155TokenId"];
+        const erc20token = event.returnValues["erc20Token"];
+        const token =
+            erc20token === "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE"
+                ? "0x0000000000000000000000000000000000000000"
+                : erc20token;
+        const entity = {
+            provider_name: this.name, // the name of the folder
+            provider_contract: this.contract, // the providers contract from which you get data
+            protocol: this.protocol,
+            nft_contract, // : this.contract,
```

Comment:
All EMVC addresses (contracts, wallets) mus be lowercase

Reply:
ok, changed to lowercase